ts_src/transaction_builder.ts

```diff
@@ -264,7 +264,52 @@ export class TransactionBuilder {
         hashType,
       );
     }
+    return signatureHash;
+  }
+
+  attachSignature(
+    vin: number,
+    keyPair: ECPairInterface,
+    signature: Buffer,
+    hashType?: number,
```

Comment:
`signatureHash` that was returned from `createSignHash` already has the hashType embedded into it.

If you use hashTypeA for `createSignHash` and hashTypeB for `attachSignature` the signature will not verify properly.

There should be a verify check here to prevent such a case.